aten/src/ATen/core/function_schema.h

```diff
@@ -147,6 +188,49 @@ struct FunctionSchema {
             is_vararg,
             is_varret) {}
 
+  // check whether this schema is backward compatible with old_schema,
+  // which means all the existing call sites of old_schema can execute with
+  // this schema. we intent to be conservative, and if necessary, we may
+  // relax the checks in future. the following conditions are considered as
+  // this schema is backward compatible with old_schema:
+  //   1) two schemas are equal
+  //   2) two schemas have same number of arguments, and this schema's
+  //      arguments are backward compatible with the corresponding ones in
+  //      argument list of old_schema.
+  //   3) this schema has m argument, old_argument has n argument, m > n.
+  //      the first n arguments of this schema are backward compatible with
+  //      the corresponding arguments of old_schema. the remaning arguments
+  //      must be either OptionalType or provide default values.
```

Comment:
OptionalType here would still be backward incompatible, because it would require that the old code pass None explicitly. The check should only be that more arguments have default values.

Reply:
good catch, removed optionaltype condition.